test/signup.test.js

```diff
@@ -0,0 +1,69 @@
+const chai = require('chai');
+const app = require('../index');
+const request = require('supertest')(app);
+const User = require('../models/User');
+const mocha = require('mocha');
+
+const describe = mocha.describe;
+const beforeAll = mocha.before
+
+const expect = chai.expect; 
+describe('POST requests', function() {
+    beforeEach(function(done){
+        User.remove({}, function(err){
+          done();
+        });
+        
+    });
+     describe('/api/users/', function(){
+        
+        describe('status code', function(){
+            it('should be equal to 201', function(done){
+                request.post('/api/users/')
+                .send({
+                    user: {
+                        email: 'goodemailformat@goodemail.com',
+                        username: 'username101',
+                        password: 'passthat1password',
+                    }
+                })
+                .expect(201, done);
```

Comment:
I suggest you use `.end()` to wrap the assertions, and these assertions should be independent of the `done` function which should be called after all your assertions.